processing/src/main/java/org/apache/druid/frame/read/columnar/DoubleFrameColumnReader.java

```diff
@@ -67,7 +67,7 @@ public ColumnPlus readColumn(final Frame frame)
     return new ColumnPlus(
         frameCol,
         ColumnCapabilitiesImpl.createSimpleNumericColumnCapabilities(frameCol.getType())
-                              .setHasNulls(NullHandling.sqlCompatible() && frameCol.hasNulls),
```

Comment:
what is the reason for this change?

Reply:
* I think even in case we are defaulting `null`-s to something somewhere; we may still have the `null` value.
running the following query with druid-27.0.0 with `druid.generic.useDefaultValueForNull: true`
```
select null,'',0,cast(null as integer)
null |  empty | 0 | null 
```
which suggests that the column value can be `null` 

* in this case the incoming column also reported that it has nulls - I think it shouldn't be allowed to narrow the scope

Reply:
the cast thing returning null is probably a bug (though its been a bug for so long that imo it isn't worth fixing in a deprecated mode), the native cast expression would spit out 0 from an expression selector if it actually ran...

I think to not risk any inadvertent changes we shouldn't make this change for now if possible. Default value mode is deprecated and will be removed in a future release, so we should probably just leave things as they are whenever possible.

Reply:
I'm just saying that in this case:
* the system do allow `null` -s - via by a simple cast - so `null` exists even in `defaultValue` mode - so I don't see it bad...
* I believe this codepath is only triggered for windowoperator related stuff - but I'll check it (I'll update this comment when it finishes)
* the incoming frame declares that it has `null` values (which is being covered)
* we also have the [ClusteredGroupPartitionerTest#testDefaultClusteredGroupPartitionerWithNulls](https://github.com/apache/druid/blob/b3015bd7ced43e3510aff22dd6f8dc9f1d4781d0/processing/src/test/java/org/apache/druid/query/rowsandcols/semantic/ClusteredGroupPartitionerTest.java#L133) which essentially inputs `null`-s as inputs...should I disable that testcase for `defaultValues` ? it will most likely just hide the issue I was facing - but I can live with that...should I just do that?

I believe the original logic incorrectly narrows to *doesn't have nulls* (`NullHandling.sqlCompatible() && frameCol.hasNulls`) - which could be `false` irrsepectively that the backing one does have (`frameCol.hasNulls`) ...if that's not allowed an exception should be thrown right away and not take further risks...

Reply:
I've taken the second approach:
* restored these classes to the previous state
* added an `assumeTrue(sqlCompatible)` to the testcase

I still think that it might not be a good idea to retain/make tweaks like this...`null` does exists in the system even when `defaultValue` is enabled...so it may naturally happen as a value in a column as well.